docs/db.md

```diff
@@ -1,4 +1,200 @@
+
+## Integration with `persistent`
+
+This is the bit that changes the most. Since we are interested in storing in our database only the `Person` entities, we are going to declare only that `Entity` using TemplateHaskell and `persistent-template`.
+
+For our specific example we are going to integrate with `persistent-sqlite`, but feel free to use whatever database you prefer! 😉
+
+```haskell
+import           Data.Int
+import qualified Data.Text               as T
+import           Database.Persist.Sqlite
+import           Database.Persist.TH
+
+mkPersist sqlSettings [persistLowerCase|
+Person json
+  name T.Text
+  age  Int32
+  deriving Show Generic
+|]
+```
+
+Notice how we are deriving `Generic` also with Persistent's QuasiQuotes.
+
+## Fixing the Id access issue
+
+If you have worked with `persistent` before, you'll know that it generates it's own Ids, and this is very convenient. In our example, we'll get for free a `PersonId` field which is what we want to get with our `PersonRequest`.
+
+But, how to derive the correct instance of `ToSchema` that `Mu` needs to work it's magic? How can we explicitly define this mapping?
+
+We have created some utilities to help you integrate with Persistent in our [`mu-persistent` package](https://github.com/higherkindness/mu-haskell/tree/master/adapter/persistent). One of such is `WithEntityNestedId`, you can use it along with `DerivingVia` to fit our needs:
+
+```haskell
+type PersonFieldMapping
+  = '[ "personAge" ':-> "age", "personName" ':-> "name" ]
+
+deriving via (WithEntityNestedId "Person" PersonFieldMapping (Entity Person))
+  instance ToSchema Maybe PersistentSchema "Person" (Entity Person)
+```
+
+Have in mind that we still need to define our own custom field mapping, in this case `PersonFieldMapping` so that the deriving via does it's job properly.
+
+## Running a pool of database connections
+
+Now let's focus on the Server!
+
+All you need to do is open one time the database, and share the connection accross all your services:
+
+```haskell
+{-# language FlexibleContexts      #-}
+{-# language OverloadedStrings     #-}
+{-# language PartialTypeSignatures #-}
+{-# language TypeApplications      #-}
+{-# language TypeFamilies          #-}
+
+module Server where
+
+import           Control.Monad.Logger
+import           Mu.GRpc.Server
+import           Mu.Server
+
+main :: IO ()
+main =
+  runStderrLoggingT $
+    withSqliteConn @(LoggingT IO) "example.db" $ \conn ->
+      liftIO $ runGRpcApp 8080 (server conn)
+```
+
+We have decided in this example to use `LoggintT` from `monad-logger` and `runStderrLoggingT` to get some basic database logs to the console for free, but this is not a must!
+
+## This actually does not work
+
+Maybe you might have noticed that this example is not going to work yet, until we tell `persistent` what is our migration. We need a small tweak in our `Schema.hs`:
```

Comment:
This is unless you have created the `example.db` file yourself. Migrations are not actually required by Persistent, are just a simple way to get a Sqlite database up and running.